tests/testthat/test-Rules-methods.R

```diff
@@ -173,3 +173,132 @@ test_that("StoppingMTDCV works correctly if CV is above threshold", {
   )
   expect_identical(result, expected) # CV is 23% > 20%.
 })
+
+
+# stopTrial-StoppingLowestDoseHSRBeta ----
+
+my_data <- h_get_data()
+
+# Model and samples are not needed to evaluate the rule, but needed for stopTrial
+# and therefore only generated once.
+my_model <- LogisticKadane(0.3, xmin = 0.001, xmax = 100)
+my_options <- McmcOptions(
+  burnin = 1, step = 1, samples = 1, rng_kind = "Mersenne-Twister", rng_seed = 94
+)
+my_samples <- mcmc(my_data, my_model, my_options)
+
+test_that("StoppingLowestDoseHSRBeta works correctly if first active dose is not toxic", {
+  stopping <- StoppingLowestDoseHSRBeta(target = 0.3, prob = 0.9)
+  result <- stopTrial(
+    stopping = stopping,
+    dose = 300,
+    samples = my_samples,
+    model = my_model,
+    data = my_data
+  )
+  expected <- structure(
+    FALSE,
+    message = paste(
+      "Probability that the lowest active dose of 25 being toxic",
+      "based on posterior Beta distribution using a Beta(1,1) prior",
+      "is 24% and thus below the required 90% threshold."
+    )
+  )
+  expect_identical(result, expected) # Prob being toxic is 24% < 90%.
+})
+
+test_that("StoppingLowestDoseHSRBeta works correctly if first active dose is toxic", {
+  stopping <- StoppingLowestDoseHSRBeta(target = 0.3, prob = 0.1)
+  result <- stopTrial(
+    stopping = stopping,
+    dose = 300,
+    samples = my_samples,
+    model = my_model,
+    data = my_data
+  )
+  expected <- structure(
+    TRUE,
+    message = paste(
+      "Probability that the lowest active dose of 25 being toxic",
+      "based on posterior Beta distribution using a Beta(1,1) prior",
+      "is 24% and thus above the required 10% threshold."
+    )
+  )
+  expect_identical(result, expected) # Prob being toxic is 24% > 10%.
+})
+
+test_that("StoppingLowestDoseHSRBeta works correctly if first active dose is not applied", {
+  my_data@x[my_data@cohort == 1] <- c(0.001, 75, 75, 75)
+  stopping <- StoppingLowestDoseHSRBeta(target = 0.3, prob = 0.1)
+  result <- stopTrial(
+    stopping = stopping,
+    dose = 300,
+    samples = my_samples,
+    model = my_model,
+    data = my_data
+  )
+  expected <- structure(
+    FALSE,
+    message = "Lowest active dose not tested, stopping rule not applied."
+  )
+  expect_identical(result, expected) # First active dose not applied.
+})
+
+my_data <- h_get_data_no_plcb()
```

Comment:
better move this in the tests. that makes it easier to run one test separately later